sql-plugin/src/main/spark330db/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastHashJoinExec.scala

```diff
@@ -147,9 +148,19 @@ case class GpuBroadcastHashJoinExec(
           GpuSemaphore.acquireIfNecessary(TaskContext.get())
         }
       }
-      val buildBatch = GpuExecutorBroadcastHelper.getExecutorBroadcastBatch(buildRelation,
+      // Create a builder which will materialize the build batch on GPU lazily.
+      val buildBatchFn = () => {
```

Comment:
I am not 100% sure. Seems the semaphore is already held (by `bufferedStreamIter.head`) when coming here. So there is no need to lazily materialize the build side batch here.

Reply:
Yes, you are right.